root/static/scripts/edit/components/FormRowTextList.js

```diff
@@ -87,21 +90,20 @@ function reducer(state: StateT, action: ActionT): StateT {
       );
 
       if (fieldCtx.read().length === 1) {
-        newStateCtx.set('field', index, 'value', '');
-        break;
+        newStateCtx.set('field', index, 'field', 'value', 'value', '');
```

Comment:
I'm probably missing something, but - shouldn't we be setting removed + the code to ensure removal even if there's only one row? I mean, I assume it works somehow because you said this is tested, but I'm confused rn.

Edit: I tested and actually it doesn't seem to work? (try to remove an ISRC from a recording with just one - the page just reloaded in place for me, no error message, no ISRC back in place, nothing)

More generally, given we're already implementing all of this to keep the removed data in the form, wouldn't it make sense to change this to have undo-able removals like we do for artist credits nowadays?

Reply:
Yeah, good catch. I found another issue where editing the field doesn't add hidden removal fields for the old value. That was tricky to fix, but I've added some Selenium tests for both of these cases. Can you try again locally?

> More generally, given we're already implementing all of this to keep the removed data in the form, wouldn't it make sense to change this to have undo-able removals like we do for artist credits nowadays?

I don't plan to implement it here, but for the future that would make sense to me.